src/scripts/mint.ts

```diff
@@ -0,0 +1,305 @@
+import { BaseError, createPublicClient, http, parseUnits } from "viem";
+import { getAllCollaterals, getCollateralInformation, mintDollar } from "./faucet";
+import {
+  allowanceButton,
+  collateralInput,
+  collateralSelect,
+  dollarInput,
+  governanceCheckBox,
+  governanceFormControl,
+  governanceInput,
+  minDollarInput,
+  mintButton,
+} from "./ui";
+import { approveToSpend, getAllowance, getTokenDecimals } from "./erc20";
+import { getConnectedClient } from "./connect-wallet";
+import { diamondAddress, dollarAddress, ubqAddress } from "./constants";
+import { ToastActions } from "./toast";
+import { mainnet } from "viem/chains";
+
+let selectedCollateralIndex = 0;
+let dollarAmount = 0;
+let dollarOutMin = 0;
+let maxCollateralIn = 0;
+let maxGovernanceIn = 0;
+let isOneToOne = false;
+
+let isButtonInteractionsDisabled = false;
+
+const collateralRecord: Record<string | number, `0x${string}`> = {};
+const toastActions = new ToastActions();
+const publicClient = createPublicClient({
+  chain: mainnet,
+  transport: http(),
+});
+
+const pathName = "mint";
+const transactionReverted = "Transaction was reverted";
+
+if (window.location.pathname.includes(pathName)) {
+  (() => {
+    setInterval(() => {
+      if (mintButton !== null) {
+        mintButton.disabled =
+          dollarAmount <= 0 ||
+          dollarOutMin <= 0 ||
+          maxCollateralIn <= 0 ||
+          (!isOneToOne && maxGovernanceIn <= 0) ||
+          typeof collateralRecord[selectedCollateralIndex] === "undefined" ||
+          isButtonInteractionsDisabled;
+      }
+
+      if (allowanceButton !== null) {
+        allowanceButton.disabled = typeof collateralRecord[selectedCollateralIndex] === "undefined" || maxCollateralIn <= 0 || isButtonInteractionsDisabled;
+      }
+
+      const appendableText = "+UBQ";
+
+      if (!isOneToOne) {
+        if (!allowanceButton.innerText.includes(appendableText)) allowanceButton.innerText = allowanceButton.innerText.concat(appendableText);
+      } else {
+        allowanceButton.innerText = allowanceButton.innerText.replace(appendableText, "");
+      }
+    }, 500);
+  })();
+
+  void (() => {
+    publicClient.watchBlocks({
+      onBlock: async () => {
+        try {
+          const collateralAddress = collateralRecord[selectedCollateralIndex];
+          const web3Client = getConnectedClient();
+
+          if (allowanceButton !== null)
+            allowanceButton.disabled = web3Client === null || !collateralAddress || !web3Client.account || isButtonInteractionsDisabled;
```

Comment:
It would be prettier if you consolidated allowanceButton and mintButton into an actionButton, since they are never displayed concomitantly and share the same objective.